dwave/cloud/client/base.py

```diff
@@ -445,12 +447,38 @@ def from_config(cls, config_file=None, profile=None, client=None, **kwargs):
         logger.debug("Creating %s.Client() with: %r", _client, config)
         return _clients[_client](**config)
 
+    class _CountDownLatch:
```

Comment:
I found this class a bit confusingly named. It looks like we currently always instantiate it with `value=0` and then inc/decrement it. I would have expected based on its name that we instantiate it with a positive `value` and then only decrement. IMO a short comment/docstring would help.

Reply:
I agree the name is confusing, will update it to something better.